### PR TITLE
[Merged by Bors] - feat(RingTheory/Algebraic): add `Algebra.IsAlgebraic.[of_]ringHom_of_comp_eq` and `iff` version

### PR DESCRIPTION
**Pull request by acmepjz**
... parallel to element-wise `IsAlgebraic.[of_]ringHom_of_comp_eq`.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by erdOne** [comment]
Thanks!
maintainer merge

**Comment by riccardobrasca** [comment]
Thanks!

bors merge

File: Mathlib/RingTheory/Algebraic.lean [hunk 1]
```diff
@@ -249,10 +249,14 @@ theorem isAlgebraic_algHom_iff (f : A →ₐ[R] B) (hf : Function.Injective f)
...
-protected theorem IsAlgebraic.ringHom_of_comp_eq [FunLike FRS R S] [RingHomClass FRS R S]
-    (f : FRS) (g : FAB) {a : A} (halg : IsAlgebraic R a)
+section
+
+variable [FunLike FRS R S] [RingHomClass FRS R S] [FunLike FAB A B] [RingHomClass FAB A B]
+  (f : FRS) (g : FAB) {a : A}
+
+theorem IsAlgebraic.ringHom_of_comp_eq (halg : IsAlgebraic R a)
     (hf : Function.Injective f)
     (h : RingHom.comp (algebraMap S B) f = RingHom.comp g (algebraMap R A)) :
```
[the remark this answers is not in the document]
Comment (acmepjz) [reply Mathlib/RingTheory/Algebraic.lean:261]: Note that `f` and `g` are only in `RingHomClass`, so here I'm extra cautious on using dot notations.

[the remark this answers is not in the document]
Comment (alreadydone) [reply Mathlib/RingTheory/Algebraic.lean:261]: Yeah, but `algebraMap _ _` is always a RingHom. I wonder whether `(algebraMap R A).comp g` works on the RHS; if yes that would be super confusing ...

**Pull request closed without merging**